src/main/commandServer/settingsValidator.ts

```diff
@@ -47,138 +58,154 @@ export default class SettingsValidator {
+          finalResult.errors.push(`Setting ${ fqname } should be a simple value, but got <${ JSON.stringify(newSettings[k]) }>.`);
         }
       } else {
         // Throw an exception if this field isn't a function, because in the verifier all values should be
         // either child objects or functions.
-        changeNeeded = allowedSettings[k].call(this, currentSettings[k], newSettings[k], errors, fqname) || changeNeeded;
+        wrapCheckCall(allowedSettings[k].call(this, currentSettings[k], newSettings[k], fqname));
       }
     }
 
-    return changeNeeded;
+    return finalResult;
+  }
+
+  protected makeErrorRecord(error: string): settingsValidatorTraits {
+    return {
+      needToUpdate:  false,
+      needToRestart: false,
+      errors:        [error]
+    };
+  }
+
+  protected makeStatusRecord(status: boolean): settingsValidatorTraits {
+    return {
+      needToUpdate:  status,
+      needToRestart: status,
+      errors:        []
+    };
   }
 
-  protected checkContainerEngine(currentValue: string, desiredEngine: string, errors: string[], fqname: string): boolean {
+  protected checkContainerEngine(currentValue: string, desiredEngine: string, fqname: string): settingsValidatorTraits {
     if (!['containerd', 'moby'].includes(desiredEngine)) {
       // The error message says 'docker' is ok, although it should have been converted to 'moby' by now.
       // But the word "'docker'" is valid in a raw API call.
-      errors.push(`Invalid value for ${ fqname }: <${ desiredEngine }>; must be 'containerd', 'docker', or 'moby'`);
-
-      return false;
+      return this.makeErrorRecord(`Invalid value for ${ fqname }: <${ desiredEngine }>; must be 'containerd', 'docker', or 'moby'`);
     }
 
-    return currentValue !== desiredEngine;
+    return this.makeStatusRecord(currentValue !== desiredEngine);
   }
 
-  protected checkEnabled(currentState: boolean, desiredState: string|boolean, errors: string[], fqname: string): boolean {
+  protected checkBoolean(currentState: boolean, desiredState: string|boolean, fqname: string): settingsValidatorTraits {
     if (typeof (desiredState) !== 'boolean') {
-      errors.push(`Invalid value for ${ fqname }: <${ desiredState }>`);
-
-      return false;
+      return this.makeErrorRecord(`Invalid value for ${ fqname }: <${ desiredState }>`);
     }
 
-    return currentState !== desiredState;
+    return this.makeStatusRecord(currentState !== desiredState);
   }
 
-  protected checkFlannel(currentState: boolean, desiredState: string|boolean, errors: string[], fqname: string): boolean {
-    if (typeof (desiredState) !== 'boolean') {
-      errors.push(`Invalid value for ${ fqname }: <${ desiredState }>`);
+  protected checkEnabled(currentState: boolean, desiredState: string|boolean, fqname: string): settingsValidatorTraits {
+    return this.checkBoolean(currentState, desiredState, fqname);
+  }
```

Comment:
It's an alias only because `checkEnabled` doesn't need to modify the result of `checkBoolean`, while `checkNoModalDialogs` does. I suppose `checkEnabled` can get pulled out, and other fields that do the same check will just map to `checkBoolean`.

Reply:
Yeah, with the splat syntax we can make `checkNoModalDialogs` simple and everything else just use `checkBoolean` directly, I think.

Reply:
I've reverted the changes to the settings validator in this PR because while they're good changes, they aren't needed for the PR's main function.